src/ctl_int.rs

```diff
@@ -17,6 +19,28 @@ const ELEM_ID_SIZE: usize = 64;
 // const ELEM_VALUE_SIZE: usize = 1224;
 // const ELEM_INFO_SIZE: usize = 272;
 
+
+/// Iterate over devices of a card.
+pub struct DeviceIter(*mut alsa::snd_ctl_t, c_int);
+
+impl DeviceIter {
+    pub(crate) fn new(ctl: *mut alsa::snd_ctl_t) -> DeviceIter {
+        DeviceIter(ctl, -1)
```

Comment:
There is a lifetime error here; you need to borrow `ctl`, not have a pointer to it, or the original ctl might be closed during the lifetime of the iterator.